pandera/decorators.py

```diff
@@ -22,6 +23,11 @@
 from .model import SchemaModel
 from .typing import AnnotationInfo
 
+if sys.version_info < (3, 7):
+    import typing_extensions as tpg
+else:
+    import typing as tpg  # pylint: disable=reimported
```

Comment:
nit: why not import `OrderedDict` directly?

Reply:
I cannot import standard library `typing` with no alias because we already import `pandera.typing`. So that's why I went for `tpg`.

Now, back to your original question. `OrderedDict` would come from`typing` in python 3.7+ and from `typing_extensions` in python 3.6. So i tried a conditional import `from ... import OrderedDict` and added some comments to make linters happy (with 3.6 and 3.7+) but then the line became too long -> formatters added brackets and split the line. But then isort and black seemed to clash, each of them adding a trailing comma to the import line, leaving the line with 2 trailing commas which is invalid code so pylint complained. I found that this way all linters were happy. Completely agree that it's not ideal.

Reply:
I'm not sure if it's worth this additional import which is used just once in this module. I think `typing.Mapping` would do just fine